File: Java/Check If All 1's Are at Least Length K Places Away.java
public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public boolean kLengthApart(int[] nums, int k) {
        int count = k;

        for (int bit : nums) {
            if (bit == 1) {
                if (count < k)
                    return false;
                count = 0;
            } else {
                count += 1;
            }
        }

        return true;
    }
}

File: Python/Check If All 1's Are at Least Length K Places Away.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def kLengthApart(self, nums: List[int], k: int) -> bool:
        count = k

        for bit in nums:
            if bit:
                if count < k:
                    return False
                count = 0
            else:
                count += 1

        return True
